arroyo-worker/src/connectors/kafka/sink/mod.rs

```diff
@@ -76,21 +102,80 @@ impl<K: Key + Serialize, T: Data + Serialize> KafkaSinkFunc<K, T> {
         format!("kafka-producer-{}", self.topic)
     }
 
-    async fn on_start(&mut self, _ctx: &mut Context<(), ()>) {
-        info!("Creating kafka producer for {}", self.bootstrap_servers);
-        let mut client_config = ClientConfig::new();
+    async fn on_start(&mut self, ctx: &mut Context<(), ()>) {
+        self.init_producer(&ctx.task_info)
```

Comment:
I believe you mentioned that transactional support needs to be enabled in the broker for this to work. If it's not, how does the error manifest? Ideally we'd get that back to the user using the error reporting system: https://github.com/ArroyoSystems/arroyo/blob/master/arroyo-worker/src/engine.rs#L471

Reply:
No, that conversation we had was just that if you're running kafka locally you need some additional settings. In docker-compose.yaml it looks like

```
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
      KAFKA_MIN_INSYNC_REPLICAS: 1
      KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
```

This converts to settings in /etc/kafka/kafka.properties like
```
offsets.topic.replication.factor=1
transaction.state.log.replication.factor=1
min.insync.replicas=1
transaction.state.log.min.isr=1
```

Transactions have been in kafka since 0.11 from 2017, so I don't think it is worth having a lot of checks for that.